python/paddle/fluid/tests/unittests/mkldnn/test_matmul_mkldnn_op.py

```diff
@@ -161,5 +161,59 @@ def set_attributes(self):
         self.attrs = {'force_fp32_output': True}
 
 
+@skip_check_grad_ci(reason="Tests inference only optimization.")
+class TestMatMulOpTransposeReshape(OpTest):
+    def generate_data(self):
+        self.x = np.random.random([1, 128, 128]).astype("float32")
+        self.y = np.random.random([1, 128, 64]).astype("float32")
+        self.out = np.matmul(self.x, self.y)
+        self.reshape_out = []
+        self.transpose_out = []
+
+    def setUp(self):
+        os.environ["DNNL_MAX_CPU_ISA"] = "AVX"
+        self.op_type = "matmul"
+        self._cpu_only = True
+        self.use_mkldnn = True
+        self.generate_data()
+
+        self.inputs = {'X': self.x, 'Y': self.y}
+        self.attrs = {'use_mkldnn': self.use_mkldnn}
+
+        if len(self.reshape_out) > 0:
+            self.attrs['fused_reshape_Out'] = self.reshape_out
+        if len(self.transpose_out) > 0:
+            self.attrs['fused_transpose_Out'] = self.transpose_out
+
+        self.inputs = {'X': self.x, 'Y': self.y}
+        self.outputs = {'Out': self.out}
+
+    def test_check_output(self):
+        self.check_output()
+
+
+class TestMatMulOpTransposeReshapeFloatWithEmpty(TestMatMulOpTransposeReshape):
+    def generate_data(self):
+        bs = 8
+        dt = "float32"
+        self.x = np.random.random([bs, 12, 128, 128]).astype(dt)
+        self.y = np.random.random([bs, 12, 128, 64]).astype(dt)
+        self.transpose_out = []
+        self.reshape_out = []
+        self.out = np.matmul(self.x, self.y)
+
+
+class TestMatMulOpTransposeReshapeFloat(TestMatMulOpTransposeReshape):
+    def generate_data(self):
+        bs = 8
+        dt = "float32"
+        self.x = np.random.random([bs, 12, 128, 128]).astype(dt)
+        self.y = np.random.random([bs, 12, 128, 64]).astype(dt)
+        self.transpose_out = [0, 2, 1, 3]
+        self.reshape_out = [0, 0, 768]
+        self.out = np.matmul(self.x, self.y).transpose([0, 2, 1, 3]).reshape(
+            [bs, -1, 768])
+
```

Comment:
A test with `int8` data type is also needed imo. This could be achieved by defining a class method
```
def init_data_type(self):
   self.data_type_ = 'float32'
```
then calling it in the setUp() method, and using self.data_type_ in the body of the `generate_data()` method instead of using the "float32" string. In the next class only override the init_data_type() method for INT8 test.

Reply:
done